src/tablestore.rs

```diff
@@ -210,6 +212,80 @@ impl TableStore {
         }
     }
 
+    #[allow(dead_code)]
+    pub(crate) async fn delete_sst(&self, id: &SsTableId) -> Result<(), SlateDBError> {
+        let path = self.path(id);
+        self.object_store
+            .delete(&path)
+            .await
+            .map_err(SlateDBError::ObjectStoreError)
+    }
+
+    pub(crate) async fn collect_garbage_wal(
+        &self,
+        min_age: Duration,
+        state: &CoreDbState,
+    ) -> Result<(), SlateDBError> {
+        let min_age = chrono::Duration::from_std(min_age).expect("invalid duration");
+        let wal_list = self
+            // Delete all SSTs up to (but not including) the last compacted WAL SST
+            .get_wal_sst_list(..state.last_compacted_wal_sst_id)
+            .await?;
+        for wal_id in wal_list {
+            let path = self.path(&SsTableId::Wal(wal_id));
+            let metadata = self.object_store.head(&path).await?;
+            if metadata
+                .last_modified
+                .signed_duration_since(chrono::Utc::now())
+                > min_age
+            {
+                self.object_store.delete(&path).await?;
+            }
+        }
+        Ok(())
+    }
+
+    pub(crate) async fn collect_garbage_ssts(
+        &self,
+        min_age: Duration,
+        state: &CoreDbState,
+    ) -> Result<(), SlateDBError> {
+        let min_age = chrono::Duration::from_std(min_age).expect("invalid duration");
+
+        // TODO Exclude snapshot SSTs from GC once snapshots are implemented
+        // Get all active SSTs
+        let l0_sst_ids: Vec<SsTableId> = state.l0.iter().map(|h| h.id).collect();
+        let compacted_sst_ids: Vec<SsTableId> = state
+            .compacted
+            .iter()
+            .flat_map(|h| h.ssts.iter().map(|h| h.id))
+            .collect();
+        let mut active_sst_id_set: HashSet<SsTableId> = HashSet::new();
+        active_sst_id_set.extend(l0_sst_ids);
+        active_sst_id_set.extend(compacted_sst_ids);
+
+        // List all SSTs in the object store
+        let sst_path = &Path::from(format!("{}/{}/", &self.root_path, self.compacted_path));
+        let mut files_stream = self.object_store.list(Some(sst_path));
+
+        // Delete all SSTs that are not in the active set and are older than min_age
+        while let Some(file) = files_stream.next().await.transpose()? {
```

Comment:
similar style/readability suggestion as above to rewrite as a `loop { match` instead of `while let`